src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/Ncwms.java

```diff
@@ -157,76 +148,61 @@ private String encodeMapForRequest(Map<String, String[]> params) {
                 sb.append('=');
                 sb.append(URLEncoder.encode(param.getValue()[0], StandardCharsets.UTF_8.name()));
             }
-        }
-        catch (Exception e) {
+        } catch (Exception e) {
             LOGGER.log(Level.SEVERE, String.format("Error encoding parameters: '%s'", e.getMessage()));
         }
 
         return sb.toString();
     }
 
-    private static List<String> getCombinedStyles(Document getCapabilitiesXml, String layerName) {
-        List<String> combinedStyles = new ArrayList<>();
-
-        DefaultXPath xpath = new DefaultXPath("//x:Layer/x:Title[.=\'" + layerName + "\']/../x:Style/x:Name/text()");
-        Map<String,String> namespaces = new TreeMap<>();
-        namespaces.put("x", "http://www.opengis.net/wms");
-        xpath.setNamespaceURIs(namespaces);
-
-        @SuppressWarnings("unchecked")
-        List<DefaultText> list = xpath.selectNodes(getCapabilitiesXml);
+    public JSONObject getMetadataJson(LayerDescriptor layerDescriptor) {
+        String wmsUrl = null;
 
-        for (final DefaultText text : list) {
-            combinedStyles.add(text.getText());
+        try {
+            wmsUrl = getWmsUrl(layerDescriptor, null);
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, String.format("Error retreiving getWmsUrl: '%s'", e.getStackTrace()));
+            e.printStackTrace();
```

Comment:
Does it make sense continuing if there's an exception here?  Perhap return null?